craft_parts/plugins/properties.py

```diff
@@ -0,0 +1,37 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2021 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Definitions and helpers for plugin options."""
+
+from typing import Any, Dict
+
+
+class PluginProperties:
```

Comment:
Should this be a dataclass?

Reply:
It has no data, but the subclasses implemented in our plugins are. Application-defined plugins have the choice to decide how to implement their property class (dataclass or not) as long as they subclass `PluginProperties`.

Reply:
should this not be an abc and unmarshal be an abstractmethod?

Reply:
The empty plugin property class is to be instantiated when a part is created without specifying a plugin properties object. (This is mainly to ensure part consistency in tests, where declaring plugin properties for each test involving part creation is very inconvenient)